### DISCUSSION TITLE
Add API for export tracks

### DISCUSSION THREAD

**Discussion by amiad** [IsraelHikingMap/Site#1794] [Q&A] [answered]
## Feature

אני רוצה לייצא מסלולים מהאתר אבל יש לי הרבה מסלולים. אשמח אם תוסיפו אפשרות לייצא מסלול דרך API ככה שאוכל בפקודה לייצא מסלוליםלפי מזהה של המסלול.

`https://israelhiking.osm.org.il/api/export?track=vjzG9bCNF5&format=gpx`

ייצא את המסלול בפורמט הרצוי.

אפשר גם ייצוא של כל המסלולים של המשתמש אבל האימות כנראה יהיה מסובך יותר.

תודה

**Comment by HarelM** [comment]
אני חושב שכבר יש אפשרות כזו. אני אבדוק וארשום פה מה המצאתי.

**Comment by amiad** [comment]
מצוין, אז אולי רק חסר תיעוד

**Comment by HarelM** [comment]
I believe you have everything you need here:
https://israelhiking.osm.org.il/swagger/index.html
You have a conversion API from the site's main data object (DataContainerPoco) to every format you would probably need, and you have a way to get the data from other APIs.

Authentication is done using OAuth2 bearer harder, you can inspect the network tab in the browser to see your token (you can place it in the swagger to get the relevant results).

I wouldn't like to advertise it more than that, so it's ok for me that this is not well documented...

**Comment by amiad** [comment, marked as answer]
ממש מגניב בדיוק מה שרציתי. הייצוא הוא דרך הכתובת
`https://israelhiking.osm.org.il/api/Urls/vjzG9bCNF5?format=gpx`

או במקרה שאני צריך פורמט של gpx single
`https://israelhiking.osm.org.il/api/Urls/vjzG9bCNF5?format=gpx_single_track`

המון תודה ולדעתי כדאי לפרסם קצת יותר, זה יכול להיות שימושי. לפחות לפרסם איך לייצא גיבוי של המסלולים של המשתמש.
